Index: a/wscript
===================================================================
--- a.orig/wscript	2017-01-26 05:03:14.250879786 +0000
+++ a/wscript	2017-01-26 05:03:52.114878924 +0000
@@ -62,6 +62,7 @@
     bld.recurse('manpages')
 
 def version(bld):
+    didGetVersion = False
     # Modified from ndn-cxx wscript
     try:
         cmd = ['git', 'describe', '--always', '--match', '%s*' % GIT_TAG_PREFIX]
